File: Greedy_Algorithm/카드 정렬하기/1715.cpp
#include <iostream>
#include <queue>

using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);  cout.tie(NULL);

    priority_queue<int, vector<int>, greater<int>> pq;
    int N;  int temp_a; int temp_b; int total = 0;
    cin >> N;
    for(int i=0; i<N; i++){
        int temp;   cin >> temp;
        pq.push(temp);
    }

    if(pq.size() != 1){
        while(true){
            temp_a = pq.top(); pq.pop();
            temp_b = pq.top(); pq.pop();
            total += temp_a + temp_b;
            pq.push(temp_a + temp_b);
            if(pq.size() == 1)
                break;

        }
    }
    
    cout << total;
}

File: Greedy_Algorithm/카드 정렬하기/1715.py
import heapq
import sys
input = sys.stdin.readline

N = int(input())
hq = []

for i in range (N):
    temp = int(input())
    heapq.heappush(hq, temp)

total = 0
temp_a = 0
temp_b = 0

if len(hq) != 1:
    while len(hq) != 0:
        temp_a = heapq.heappop(hq)
        temp_b = heapq.heappop(hq)
        total += temp_a + temp_b
        if (len(hq)) == 0:
            break
        heapq.heappush(hq, temp_a + temp_b)

print(total)